MapboxNavigation/NavigationOptions.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import MapboxCoreNavigation
+
+/*
+    Object representing available options for a Navigation Session.
+ */
+
+@objc(MBNavigationOptions)
+open class NavigationOptions: NSObject {
+    
+    /*
+     The styles that the view controller’s internal `StyleManager` object can select from for display. Defaults to `[DayStyle(), NightStyle()]`
+     */
+    @objc public var styles: [Style]? = nil
+    
+    /*
+     The navigation service that manages navigation along the route.
+     */
+    @objc open var navigationService: NavigationService?
+    
+    /*
+     The voice controller that manages the delivery of voice instructions during navigation.
+    */
+    @objc public var voiceController: RouteVoiceController?
+    
+    /*
+     The view controlelr to embed into the bottom section of the UI. Defaults to `BottomBannerViewController`.
+     */
+    @objc public var bottomBanner: ContainerViewController?
+    
+    @objc public convenience init(styles: [Style]? = nil, navigationService: NavigationService? = nil, voiceController: RouteVoiceController? = nil, bottomBanner: ContainerViewController? = nil) {
```

Comment:
As a developer, I would be a little surprised that CarPlayManager and CarPlayNavigationViewController initializers continue to take a long list of arguments instead of a NavigationOptions. There’s some interesting overlap but also some significant differences between the sets of options that each environment would require. It’s fine if we make the CarPlay classes adopt NavigationOptions later on, but any properties in common between the two environments should probably live in a protocol to start out with.